test/test_custom_ops.py

```diff
@@ -402,19 +407,20 @@ def test_opcheck_opinfo(self, device, dtype, op):
             torch.library.opcheck(op.op, args, kwargs)
 
     def test_opcheck_fails_basic(self, device):
-        @custom_op(f"{self.test_ns}::foo")
-        def foo(x: torch.Tensor) -> torch.Tensor: ...
+        lib = self.lib()
+        lib.define("foo(Tensor x) -> Tensor")
+        op = self.ns().foo.default
 
-        @foo.impl(["cpu", "cuda"])
         def foo_impl(x):
             return x.sum()
 
+        lib.impl("foo", foo_impl, "CPU")
+        lib.impl("foo", foo_impl, "CUDA")
+        lib.impl("foo", foo_impl, "XPU")
+
         x = torch.randn(3, device=device, requires_grad=True)
-        # Triggers the CustomOp autograd NYI error
-        with self.assertRaisesRegex(
-            optests.OpCheckError, "Autograd has not been implemented for operator"
-        ):
-            torch.library.opcheck(self.get_op(f"{self.test_ns}::foo"), (x,), {})
```

Comment:
Does `torch.library.opcheck` not support XPU? This UT is used to check `torch.library.opcheck`, now it is changed.

Reply:
> Does `torch.library.opcheck` not support XPU? 

As it occurs, it does not support XPU. This PR enables a number of tests for XPU, but as I noted in PR description more work is needed as there is a mixture of tests for and relying on the deprecated functionality which was not cleaned up: "Tests are using `torch._custom_ops` deprecated API planned for removal at pytorch 2.6 (not done)". That's where a comment and guidance from maintainers would be nice on the next step. I suggest to consider this PR for merge and a follow up to first actually remove `torch._custom_ops` from library and tests and then add XPU support to the remainder of supported features. @albanD